### PR TITLE
Disables JSON formatting on client

### LINKED ISSUE #508: Polish form behavior
Still have a problem in entering the multiline config:

1. JSON to text convert has no space after `,`
2. Clicks on `"browserslist": [` has different results
3. Config with just comment has unclear `Audience coverage: 0.00 %` output (should be equal to empty field)

### PR DESCRIPTION
**Pull request by tatarianBarbarian**
#508 

Had a discussion with @ai on the ways of solving JSON-related issues.

I suggest turning off client-side transforming JSON to query.

The reason is simple: when a user comes to the site with config in a certain format(JSON|RC), they mostly want output in the same format as input.

I have moved the processing code to the server, so JSON input will be processed correctly.

For now, the only case when JSON can be transformed to a query format is applying fixes from linter.

https://github.com/browserslist/browsersl.ist/assets/35267898/8df7de1d-52e1-48e2-a308-b8ab40a0bb4b

### CODE REVIEW

File: server/lib/parse-config.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    return JSONToQuery(config)
+  } catch {}
+  try {
+    return JSONFragmentToQuery(config)
+  } catch {}
+
+  return config
+}
+
+function rcConfigToQuery(config) {
```
Comment (ai) [inline server/lib/parse-config.js:24]: `rc` means text config?

Let’s use `textConfigToQuery` instead.

File: server/lib/parse-config.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  return config
+    .toString()
+    .split(/\n|,/)
+    .map(filterComments)
+    .filter(line => line !== '')
+}
+
+function JSONToQuery(jsonConfig) {
```
Comment (ai) [inline server/lib/parse-config.js:32]: Inconsistency with `jsonConfigToQuery`. Let’s use `jsonToQuery`

File: server/lib/parse-config.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  } catch {}
+
+  return config
+}
+
+function rcConfigToQuery(config) {
+  return config
+    .toString()
+    .split(/\n|,/)
+    .map(filterComments)
```
Comment (ai) [inline server/lib/parse-config.js:28]: It is faster to call `filterComments` on `config.toString()`

File: server/lib/parse-config.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    .filter(line => line !== '')
```
Comment (ai) [inline server/lib/parse-config.js:29]:
```suggestion
    .filter(line => line.trim().length > 1)
```

**Comment by ai** [comment]
Let's also add `\n` to `"browserslist": [` button

File: server/test/parse-config.test.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  {
+    "private": true,
+    "browserslist": [
+      "IE > 9",
+      "chrome > 80"
+    ]
+  }
+  `
+
+  equal(configToQuery(input), 'IE > 9,chrome > 80')
```
Comment (ai) [inline server/test/parse-config.test.js:28]:
```suggestion
  equal(configToQuery(input), 'IE > 9, chrome > 80')
```

**Comment by a-chabin** [comment]
I like that we keep the original formatting.
It doesn't look pretty in the address bar. But it's cool that now you can share your original config via the link.

**Comment by a-chabin** [comment]
@ai 
@tatarianBarbarian 

Always when copying from package.json we get extra spaces.

Maybe we need to remove extra spaces from the address bar and from the request to the API?
I propose not modify the user input.

image

**Comment by ai** [comment]
> Maybe we need to remove extra spaces from the address bar and from the request to the API?

Not now. With these extra spaces it is easy to copy query back.

**Review by a-chabin — APPROVED** [review]

**Pull request merged**